File: app/src/main/java/com/frame/activity/KnowledgeHierarchyActivity.java
package com.frame.activity;

import android.content.Intent;
import android.os.Bundle;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;

import com.blankj.utilcode.util.ActivityUtils;
import com.blankj.utilcode.util.ObjectUtils;
import com.frame.R;
import com.frame.common.CommonData;
import com.frame.dataclass.DataClass;
import com.frame.observers.ProgressObserver;
import com.frame.utils.CU;
import com.frame.utils.JU;
import com.google.gson.internal.LinkedTreeMap;
import com.scwang.smartrefresh.layout.SmartRefreshLayout;

import org.byteam.superadapter.SuperAdapter;
import org.byteam.superadapter.SuperViewHolder;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.OnClick;

/**
 */
public class KnowledgeHierarchyActivity extends BaseTitleActivity {

    @BindView(R.id.srlCommon)
    SmartRefreshLayout mSmartRefreshLayout;
    @BindView(R.id.rvCommon)
    RecyclerView mRecyclerView;

    private SuperAdapter mSuperAdapter;
    private List<LinkedTreeMap<String, Object>> mList = new ArrayList<>();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.common_layout_srl_rv);
        initControl();
    }

    private void initControl() {
        setTitleText("知识体系");

        mRecyclerView.setLayoutManager(new LinearLayoutManager(mBActivity));
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setAdapter(mSuperAdapter = getSuperAdapter());

        setSmartRefreshLayout();
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (ObjectUtils.isEmpty(mList)) {
            getNetData();
        }
    }


    @OnClick(R.id.llGotoTop)
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.llGotoTop:
                mRecyclerView.smoothScrollToPosition(0);
                break;
        }
    }

    private void setSmartRefreshLayout() {
        mSmartRefreshLayout.setOnRefreshListener(refreshLayout -> {
            getNetData();
        });
    }

    @SuppressWarnings("unchecked")
    private SuperAdapter getSuperAdapter() {
        return new SuperAdapter<LinkedTreeMap<String, Object>>(mBActivity, mList, R.layout.item_knowledge_hierarchy) {
            @Override
            public void onBind(SuperViewHolder holder, int viewType, int layoutPosition, LinkedTreeMap<String,
                    Object> map) {
                holder.setText(R.id.tvKnowledgeHierarchyTitle, JU.s(map, "name"))
                        .setTextColor(R.id.tvKnowledgeHierarchyTitle, CU.randomColor());

                StringBuilder content = new StringBuilder();
                List<LinkedTreeMap<String, Object>> list = JU.al(map, "children");
                for (LinkedTreeMap<String, Object> ltm : list) {
                    content.append(JU.s(ltm, "name")).append("   ");
                }
                holder.setText(R.id.tvKnowledgeHierarchyContent, content);
                holder.itemView.setOnClickListener(view -> {
                            ActivityUtils.startActivity(new Intent(mBActivity, WxArticleActivity.class)
                                    .putExtra(CommonData.PARAM1, map));
                        }
                );
            }
        };
    }

    private void getNetData() {
        BaseActivity.doCommonGet("tree/json", null, new ProgressObserver<DataClass>(mBActivity, true,
                mSmartRefreshLayout) {
            @Override
            public void onNext(DataClass dc) {
                mList.clear();
                mList.addAll(JU.al(dc.object, "data"));
                mSuperAdapter.notifyDataSetChanged();
            }
        });
    }


}

File: app/src/main/java/com/frame/activity/KnowledgeHierarchyActivity.kt
package com.frame.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import com.blankj.utilcode.util.ObjectUtils
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.frame.R
import com.frame.common.CommonData
import com.frame.dataclass.DataClass
import com.frame.observers.ProgressObserver
import com.frame.utils.CU
import com.frame.utils.JU
import com.frame.utils.LU
import com.google.gson.internal.LinkedTreeMap
import kotlinx.android.synthetic.main.common_layout_srl_rv.*
import java.util.*

/**
 */
class KnowledgeHierarchyActivity : BaseTitleActivity() {

    private var mQuickAdapter: BaseQuickAdapter<LinkedTreeMap<String, Any>, BaseViewHolder>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.common_layout_srl_rv)
        initControl()
    }

    private fun initControl() {
        setTitleText("知识体系")
        recyclerView?.run {
            layoutManager = LinearLayoutManager(mBActivity)
            setHasFixedSize(true)
            adapter = getQuickAdapter().also { mQuickAdapter = it }
        }
        setSmartRefreshLayout()
    }

    override fun onResume() {
        super.onResume()
        if (ObjectUtils.isEmpty(mQuickAdapter?.data)) getNetData()
    }

    fun onViewClicked(view: View) {
        when (view.id) {
            R.id.ivGotoTop -> recyclerView?.smoothScrollToPosition(0)
        }
    }

    private fun setSmartRefreshLayout() {
        refreshLayout?.setOnRefreshListener { getNetData() }
    }

    private fun getQuickAdapter(): BaseQuickAdapter<LinkedTreeMap<String, Any>, BaseViewHolder> {
        return object : BaseQuickAdapter<LinkedTreeMap<String, Any>, BaseViewHolder>(R.layout.item_knowledge_hierarchy) {
            override fun convert(holder: BaseViewHolder, map: LinkedTreeMap<String, Any>) {
                holder.setText(R.id.tvKnowledgeHierarchyTitle, JU.s(map, "name"))
                    .setTextColor(R.id.tvKnowledgeHierarchyTitle, CU.randomColor())
                val content = StringBuilder()
                val list = JU.al<List<LinkedTreeMap<String, Any>>>(map, "children")
                for (ltm in list) {
                    content.append(JU.s(ltm, "name")).append("   ")
                }
                holder.setText(R.id.tvKnowledgeHierarchyContent, content)
                holder.itemView.setOnClickListener { view: View? ->
                    LU.gotoActivityAnim(view, Intent(mBActivity, WxArticleActivity::class.java)
                        .putExtra(CommonData.PARAM1, map))
                }
            }
        }
    }

    private fun getNetData() {
        doCommonGet("tree/json", null, object : ProgressObserver<DataClass>(mBActivity, true, refreshLayout) {
            override fun onNext(dc: DataClass) {
                mQuickAdapter?.setList(JU.al<ArrayList<LinkedTreeMap<String, Any>>>(dc.obj, "data"))
            }
        })
    }
}